lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -1550,6 +1567,15 @@ class ConvertAtenMatmulBaseOp : public OpConversionPattern<AtenOpT> {
                 matmulLhs, matmulRhs)
             .getResult();
 
+    auto originalMatMulInputType = lhsElemTy;
+    auto castOpResult =
+        rewriter
+            .create<tosa::CastOp>(op->getLoc(),
```

Comment:
```suggestion
            .createOrFold<tosa::CastOp>(op->getLoc(),
```
to avoid the cast if the type is the same

Reply:
done